File: 프로그래머스/Level1/로또의 최고순위와 최저순위.js



function solution(lottos, win_nums) {
    var answer = [];
    var rank =[6,6,5,4,3,2,1]
    var Cnt = 0;
    var Zero = 0;
    for (let i=0; i<lottos.length; i++){
        if (win_nums.includes(lottos[i])){
            Cnt += 1;
        }
        if (lottos[i] ==0){
            Zero += 1;
        }
    }
    
    
    answer[0] = rank[Cnt + Zero]
    answer[1] = rank[Cnt]
    return answer;
}

File: 프로그래머스/Level1/로또의 최고 순위와 최저순위.py


def solution(lottos, win_nums):
    answer = []
    rates = [6,6,5,4,3,2,1]
    zero = 0
    win = 0
    for i in range(len(lottos)):
        if lottos[i] == 0:
            zero += 1
        elif lottos[i] in win_nums:
            win += 1
    
    answer.append(rates[win+zero])
    answer.append(rates[win])
    return answer